packages/apps/spaces/app/organization/[id]/src/components/Timeline/FutureZone/reminders/Reminders.tsx

```diff
@@ -1,61 +1,132 @@
+  const updateReminder = useUpdateReminderMutation(client, {
+    onMutate: (values) => {
+      queryClient.cancelQueries({ queryKey: remindersQueryKey });
+
+      const previousEntries = useRemindersQuery.mutateCacheEntry(queryClient, {
+        organizationId,
+      })((cache) =>
+        produce(cache, (draft) => {
+          if (!draft) return;
+
+          const foundReminder = draft.remindersForOrganization.find(
+            (r) => r.metadata.id === values.input.id,
+          );
+
+          if (!foundReminder) return;
+          foundReminder.content = values.input.content ?? '';
+          foundReminder.dismissed = values.input.dismissed ?? false;
+
+          setTimelineMeta((prev) =>
+            produce(prev, (draft) => {
+              draft.reminders.recentlyUpdatedId = values.input.id ?? '';
+            }),
+          );
+        }),
+      );
+
+      return { previousEntries };
+    },
+    onError: (_, __, context) => {
+      if (context?.previousEntries) {
+        queryClient.setQueryData(remindersQueryKey, context.previousEntries);
+      }
+      toastError(`We couldn't update the reminder`, 'update-reminder-error');
+    },
+    onSettled: () => {
+      setTimeout(() => {
+        queryClient.invalidateQueries({ queryKey: remindersQueryKey });
+      }, 500);
     },
   });
 
-  if (isPending) return <p>Loading...</p>;
+  const onChange = (values: ReminderEditForm) => {
+    updateReminder.mutate({
+      input: {
+        id: values.id,
+        dueDate: values.date,
+        content: values.content,
+      },
+    });
+  };
+
+  const onDismiss = (id: string) => {
+    updateReminder.mutate({
+      input: {
+        id: id,
+        dismissed: true,
+      },
+    });
+  };
+
+  const { data: globalCacheData } = useGlobalCacheQuery(client);
+
+  const user = globalCacheData?.global_Cache?.user;
+  const currentOwner = [user?.firstName, user?.lastName]
+    .filter(Boolean)
+    .join(' ');
+
+  if (isPending) return null;
 
   return (
     <VStack align='flex-start'>
-      {data?.map((r) => (
-        <ReminderPostit key={r.id}>
-          <Flex px='4' mb='2'>
-            <Text>{r.content}</Text>
-          </Flex>
-          <Flex align='center' px='4' w='full' justify='space-between' mb='2'>
-            <Text>24 Mar • 09:09</Text>
-            <Button variant='ghost' colorScheme='yellow' size='sm'>
-              Dismiss
-            </Button>
-          </Flex>
-        </ReminderPostit>
-      ))}
+      {data?.remindersForOrganization
+        ?.filter((r) => !r.dismissed)
+        .sort(
+          (a, b) =>
+            new Date(a?.dueDate).valueOf() - new Date(b?.dueDate).valueOf(),
+        )
+        .map((r) => (
+          <ReminderItem
+            key={r.metadata.id}
+            currentOwner={currentOwner}
+            onChange={onChange}
+            onDismiss={onDismiss}
+            data={mapReminderToForm(r)}
+          />
+        ))}
     </VStack>
   );
 };
+
+function mapReminderToForm(
```

Comment:
could that be DTO?

Reply:
It coulsd, but I feel that it's a bit overkill.